### ISSUE TITLE
If I double-click on a sample track element, LMMS should show me MySamples first

### ISSUE THREAD

**Issue by raindropsfromsky** [LMMS/lmms#4221]
If I double-click on a sample track element, it shows me this collection:

![image](https://user-images.githubusercontent.com/9047168/37015191-5c4d4508-212b-11e8-9df8-8d44984ccd79.png)

Insread, LMMS should show me the **MySamples** collections first, because the most suitable samples are already bundled with LMMS.

**Comment by Spekular** [comment]
>the most suitable samples are already bundled with LMMS

I disagree, and don't see what your argument for this is. Any samples
someone has collected on their own are probably more interesting to them
than the default. Otherwise, why would they get them in the first place?

**Comment by raindropsfromsky** [comment]
I am not denying the importance of self-collection. I am not recomending that they should be removed.

My point is that the default collection is not offered. For newbies that is sufficient.

**Comment by tresf** [comment]
The sample track has most usefulness for custom samples.  One could argue a stock useless sample is better than an empty folder (valid point), but the downside to this approach is one would have to change his/her preferences in order to get it to eventually open to the `LMMS WORKING DIR/samples` folder which would only make this harder on the user.

But opening samples this way is archaic and we need to embrace drag/drop anyway.  So... what we **should** do is the following:

1. Enable drag and drop of samples into the Song Editor and Beat/Bassline Editor (well, we already support this... so, really just the second point... )
2. Any sample dragged/dropped in over a certain length should open in sample track by default with an option to override this (perhaps through context menu).

But what this exposes is historic UI that we're stuck with until someone codes it better.  This isn't a new conversation and it's not worth of its own bug report without clear action (and what is proposed is a strange band-aid on a bigger problem).

The bottom line is sample track lacks some of the most basic features which would make it intuitive and useful.  I feel I'm a good judge of this, because I make use of it heavily.  Functionality is slowly getting better.  Closing marking as duplicate of https://github.com/LMMS/lmms/issues/1471.

**Comment by Spekular** [comment]
"Default samples are sufficient for newbies"

How does this statement justify worsening the workflow for everyone who
uses third party samples (which I would assume is the vast majority of
those who are past the absolute newbie stage)?

**Comment by Spekular** [comment]
"The default collection is not offered"

This is false. You can browse to them if you insist on using double click,
or you could drag them from the sidebar where factory samples are easily
available.

**Comment by tresf** [comment]
> worsening the workflow for everyone who uses third party samples

Correct.  We really just need to move away from the common dialog selection boxes for default behaviors.  There are still people inserting VSTs this way because they've grown accustomed to it and it's several steps more than dragging a `.dll`.  @raindropsfromsky's points are valid, they're just out of scope for the limited workflow options we offer, unfortunately at the benefit of non-newcomers, sorry. :/

**Comment by Spekular** [comment]
@tresf why not just allow samples to be dragged onto sample tracks? I'm not
a fan of some arbitrary track length being opened as a sample track rather
than in an AFP. In fact, in some projects I've loaded loops in the AFP.

Benefits over sample tracks:

-You can pitch them with the built in pitch knob
-You can pitch them by playing different notes
-You can reverse them
-You can easily set start/end/loop points, loop mode
-You can set them to start/stop at 16ths/32nds etc rather than using the
finicky drag+ctrl in the song editor
-Set envelopes/lfos

I realize all of these could in theory be dealt with by improving sample
tracks (vastly), but I don't see that happening in the near future. At that
point it may be an idea to merge AFP and sample tracks altogether, which
would make it a non-issue to decide what track to create for dragged
samples.

**Comment by Spekular** [comment]
Seems my formatting put most of my comment under the ellipsis together with the email junk that I wish github would just strip altogether.

**Comment by tresf** [comment]
The formatting is due to you using email instead of GitHub.

> why not just allow samples to be dragged onto sample tracks?

I don't think anyone said we wouldn't.

>  I'm not a fan of some arbitrary track length being opened as a sample track rather
than in an AFP

It wouldn't really be arbitrary, it would be a sane default but I understand the point.

Anyway, commercial daws usually have a quick context menu for converting, which would help the end-user reverse this problem.

> Benefits over sample tracks:

Yeah, we know. :)

**Comment by BaraMGB** [comment]
Just for the record: https://github.com/LMMS/lmms/issues/3228
